ui/app/selectors/selectors.js

```diff
@@ -211,9 +211,14 @@ function conversionRateSelector (state) {
 
 function getAddressBook (state) {
   const network = state.metamask.network
-  const addressBookEntries = Object.values(state.metamask.addressBook)
-    .filter(entry => entry.chainId && entry.chainId.toString() === network)
+  let addressBookEntries = []
+  addressBookEntries = Object.keys(state.metamask.addressBook)
+    .filter(entry => entry && entry === network)
+  if (addressBookEntries.length !== 0) {
 
+    addressBookEntries = Object.values(addressBookEntries)
+    return Object.values(state.metamask.addressBook[addressBookEntries])
```

Comment:
I think this section could be replaced by something like this:
```
function getAddressBook (state) {
  const network = state.metamask.network
  return state.metamask.addressBook[network] || {}
```

The current implementation is doing some pretty weird things that I'm assuming were accidental. `addressBookEntries` for example is an array of chainIds, not the actual entries. So the subsequent `Object.values` call is a no-op, and the final return statement only works because of some weird quirk of JavaScript that I just learned of today 😅  (That `['fool`] gets coerced to `'foo'` when used as an index, e.g. `obj[['foo']] === obj['foo']` )

I'm also not sure we'd want to return the full address book if the current network had no address book entries. Returning an empty object would seem more appropriate.

Reply:
Sorry, I edit the above comment about a million times, I had misread part of this code and I made many typos. I think it's correct now though.

Reply:
good call, updated to simplified example